Engine/DataFeeds/SubscriptionDataReader.cs

```diff
@@ -284,8 +284,15 @@ public void Initialize()
 
             _delistingDate = _config.Symbol.GetDelistingDate(_mapFile);
 
-            // adding a day so we stop at EOD
-            _delistingDate = _delistingDate.AddDays(1);
+            var isFutureOrFutureOption = _config.Symbol.SecurityType == SecurityType.Future ||
+                _config.Symbol.SecurityType == SecurityType.FutureOption;
+
+            if (!isFutureOrFutureOption)
+            {
+                // adding a day so we stop at EOD for other asset classes except futures and futures options,
+                // since they expire on the same day as their expiry during market hours.
+                _delistingDate = _delistingDate.AddDays(1);
```

Comment:
Without this change, I believe futures and FOPs will report the day after their expiry as a tradable date since we're not comparing with >= here: https://github.com/QuantConnect/Lean/blob/b432f63ac7ffca53042ff546cf544a5b00234a15/Engine/DataFeeds/SubscriptionDataReader.cs#L575

Reply:
> I believe futures and FOPs will report the day after their expiry as a tradable date

The day after is the break condition of the loop, I think it has no effect. Reverting this change all test still pass, don't think it's required? Suggest we revert it if so